client/my-sites/stats/features/modules/stats-utm/stats-module-utm.jsx

```diff
@@ -43,12 +45,53 @@ const StatsModuleUTM = ( {
 	query,
 	postId,
 	summaryUrl,
+	context,
 } ) => {
 	const siteId = useSelector( getSelectedSiteId );
 	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
 	const translate = useTranslate();
+	const [ selectedOption, setSelectedOption ] = useState( () => {
+		const utmQueryParam = context.query[ UTM_QUERY_PARAM ];
+		return Object.values( OPTION_KEYS ).includes( utmQueryParam )
+			? utmQueryParam
+			: OPTION_KEYS.SOURCE_MEDIUM;
+	} );
+	const queryParams = useMemo( () => {
+		let urlParams;
+
+		if ( summaryUrl ) {
+			urlParams = new URLSearchParams( summaryUrl?.split( '?' )[ 1 ] || '' );
+		} else {
+			urlParams = new URLSearchParams( context.query );
+		}
+
+		return urlParams;
+	}, [ summary, summaryUrl, context.query ] );
```

Comment:
```suggestion
	}, [ summaryUrl, context.query ] );
```
`summary` is not the dependency.